### PR TITLE
Allow CoreWidgetProvider to be disposed

### LINKED ISSUE #3411: Closing dev home does not terminate the dev home process
### Dev Home version

_No response_

### Windows build number

_No response_

### Other software

_No response_

### Steps to reproduce the bug

1. Launch Dev Home.
2. Close the window.

### Expected result

Dev Home is terminated. 

### Actual result

Dev Home process continue to run. Core Widget Provider process keeps dev home around.

### Included System Information

_No response_

### Included Extensions Information

_No response_

### PR DESCRIPTION
**Pull request by krschau**
## Summary of the pull request
When Dev Home's ExtensionService starts an extension process, it keeps a reference around so that when Dev Home is closed, it can signal to the extension that it can stop. Previously, we have relied on the WidgetService to start the CoreWidgetProvider process. Since it implements no other Dev Home provider, nowhere inside Dev Home was starting it, so we had no way to signal it to stop and it would continue running.

This change starts the CoreWidgetProvider as a Dev Home extension when a core widget it put on the Dashboard (either via Add Widget or having been pinned during a previous run). The WidgetExtensionService is a new service that handles this. Now it can be signaled to stop and does when Dev Home closes.

Core Widgets can still be used in the Windows Widget Board, the WidgetService can still start the process when needed there.

## Detailed description of the pull request / Additional comments
Additional changes:
* In ExtensionService, removed an unused method, made a method that was only used locally private, and added a new method that returns an ExtensionWrapper given an extension's unique ID
* In ExtensionWrapper, if we get a bad HRESULT throw it right away instead of trying to use the bad object.
* ExtensionLibraryViewModel had a method with the same name as one from the ExtensionService. Changed the name of the method in ExtensionLibraryViewModel just to keep things more separate.

## Validation steps performed
Validated locally using Dev and Canary core widgets.

## PR checklist
- [ ] Closes #3411
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: src/Services/ExtensionService.cs [hunk 1]
```diff
@@ -227,6 +223,12 @@ public async Task<IEnumerable<IExtensionWrapper>> GetInstalledExtensionsAsync(bo
         }
     }
 
+    public IExtensionWrapper? GetInstalledExtension(string extensionUniqueId)
+    {
+        var extension = _installedExtensions.Where(extension => extension.ExtensionUniqueId == extensionUniqueId);
```
Comment (dhoehna) [inline src/Services/ExtensionService.cs:228]: Please use string.Equals().

File: src/Services/ExtensionService.cs [hunk 2]
```diff
@@ -227,6 +223,12 @@ public async Task<IEnumerable<IExtensionWrapper>> GetInstalledExtensionsAsync(bo
         }
     }
 
+    public IExtensionWrapper? GetInstalledExtension(string extensionUniqueId)
+    {
+        var extension = _installedExtensions.Where(extension => extension.ExtensionUniqueId == extensionUniqueId);
+        return extension.First() ?? null;
```
Comment (dhoehna) [inline src/Services/ExtensionService.cs:229]: NIT: .FirstOrDefault()  will return null if the collection is empty.

File: tools/Dashboard/DevHome.Dashboard/Services/IWidgetExtensionService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+
+namespace DevHome.Dashboard.Services;
+
+internal interface IWidgetExtensionService
+{
+    bool IsCoreWidgetExtension(string providerDefinitionId);
```
Comment (dhoehna) [inline tools/Dashboard/DevHome.Dashboard/Services/IWidgetExtensionService.cs:10]: I don't understand this name.  Is Core Widget Extension "what"?  Deleted, live, something else?  Or is the question "IS the core widget an extension? (Raised voice because I have no idea)

**Review by dhoehna — APPROVED** [review]

File: tools/Dashboard/DevHome.Dashboard/Services/IWidgetExtensionService.cs (hunk 3 above)
Comment (krschau) [reply tools/Dashboard/DevHome.Dashboard/Services/IWidgetExtensionService.cs:10]: It's an awkward place because we're talking about this *thing* both as a widget provider and a Dev Home extension and tying the two together. Looking at this method again, I didn't need to introduce that complexity to this particular method so I'm changing the name to IsCoreWidgetProvider (I think IsProviderACoreWidgetProvider is just too wordy) and between that plus a comment plus now the name correlating to the parameter name hopefully makes it clear enough.

**Review by dkbennett — APPROVED** [review]
